clients/vault/src/oracle/storage/impls.rs

```diff
@@ -153,6 +156,75 @@ impl FileHandler<TxHashMap> for TxHashesFileHandler {
 	}
 }
 
+impl ScpAchiveStorage{
+
+	pub async fn get_scp_archive(slot_index : i32) -> Result<XdrArchive::<ScpHistoryEntry>, Error>{
+		let (url, file_name) = Self::get_url_and_file_name(slot_index);
+		let result = Self::gz_decode_file(&file_name);
+
+		if result.is_err(){
+			let result = Self::download_file_and_save(&url, &file_name).await;
+			if result.is_ok(){
+				let data = Self::gz_decode_file(&file_name)?;
+				return Ok(Self::decode_xdr(data));
+			}
+		}
```

Comment:
Why are you calling `Self::gz_decode_file(&file_name);` twice? If this is necessary maybe add some comments as to why and what is happening because it is not obvious.

Reply:
Done